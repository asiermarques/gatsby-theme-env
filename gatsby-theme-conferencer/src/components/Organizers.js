import React from "react";

const Organizers = ({ organizers }) => (
  <>
    <h3>Organizers</h3>
    <div className="row justify-content-start">
      {organizers.length &&
        organizers.map((organizer, index) => (
          <div key={index} className="col-6 col-md-3">
            <a
              href={organizer.link}
              title={organizer.name}
              target="_blank"
              rel="noopener noreferrer"
            >
              <img
                src={organizer.image}
                alt={organizer.name}
                className="img-thumbnail"
              />
            </a>
          </div>
        ))}
    </div>
  </>
);

export default Organizers;
