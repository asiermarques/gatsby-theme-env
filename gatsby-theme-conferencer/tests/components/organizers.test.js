import React from "react";
import { describe, it } from "@jest/globals";
import renderer from "react-test-renderer";

import Organizers from "../../src/components/Organizers";
import OrganizersConfigFixture from "../__fixtures/OrganizersConfig";

describe("Organizers", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<Organizers organizers={OrganizersConfigFixture} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
