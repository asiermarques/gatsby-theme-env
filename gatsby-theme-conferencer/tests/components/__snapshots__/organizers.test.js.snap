// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Organizers renders correctly 1`] = `
Array [
  <h3>
    Organizers
  </h3>,
  <div
    className="row justify-content-start"
  >
    <div
      className="col-6 col-md-3"
    >
      <a
        href="https://asiermarques.com"
        rel="noopener noreferrer"
        target="_blank"
        title="Sara Smith"
      >
        <img
          alt="Sara Smith"
          className="img-thumbnail"
          src="https://randomuser.me/api/portraits/women/40.jpg"
        />
      </a>
    </div>
    <div
      className="col-6 col-md-3"
    >
      <a
        href="https://asiermarques.com"
        rel="noopener noreferrer"
        target="_blank"
        title="Sara Smith"
      >
        <img
          alt="Sara Smith"
          className="img-thumbnail"
          src="https://randomuser.me/api/portraits/women/40.jpg"
        />
      </a>
    </div>
    <div
      className="col-6 col-md-3"
    >
      <a
        href="https://asiermarques.com"
        rel="noopener noreferrer"
        target="_blank"
        title="Sara Smith"
      >
        <img
          alt="Sara Smith"
          className="img-thumbnail"
          src="https://randomuser.me/api/portraits/women/40.jpg"
        />
      </a>
    </div>
  </div>,
]
`;
